index.js

```diff
@@ -849,24 +850,29 @@ RingPop.prototype.handleOrProxyAll =
         }
 
         dests.forEach(function(dest) {
+            var destKeys = keysByDest[dest];
             var res = hammock.Response(function(err, resp) {
                 onResponse(err, resp, dest);
             });
             if (whoami === dest) {
                 self.logger.trace('handleOrProxyAll was handled', {
-                    keys: keys,
+                    keys: destKeys,
                     url: req && req.url,
                     dest: dest
                 });
-                self.emit('request', req, res);
+                var head = serializeHead(req, {
```

Comment:
you'll be unnecessarily stringifying head and then having to parse again. Port what I did in this PR over to yours and you can use `rawHead`: https://github.com/uber/ringpop/pull/41/files#diff-d89c23da890a7f6cf4d117b1f209e9dcR22